src/services/data-targets/fiware-data-target.service.ts

```diff
@@ -77,28 +119,31 @@ export class FiwareDataTargetService extends BaseDataTargetService {
                 config.authorizationType === AuthorizationType.HEADER_BASED_AUTHORIZATION
             ) {
                 axiosConfig.headers["Authorization"] = config.authorizationHeader;
+            } else if (
+                config.authorizationType === AuthorizationType.OAUTH_AUTHORIZATION
+            ) {
+                const token = await this.authenticationTokenProvider.getToken(config)
+                axiosConfig.headers["Authorization"] = `Bearer ${token}`;
```

Comment:
token can be `undefined`

Reply:
In what case? `getToken` returns token from cache and if there is no such then it's requesting new one. Am I missing something?

Reply:
On error, the method will hit the catch-clause and log the error. No explicit return is made to `undefined` is returned.

Reply:
Rerthrowing error in getToken so it will never return undefined